lib/services/expiry/expiry.go

```diff
@@ -0,0 +1,250 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package expiry
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils/interval"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+)
+
+var (
+	// scanInterval is the interval at which the expiry checker scans for access requests
+	scanInterval              = time.Minute * 5
+	pendingRequestGracePeriod = time.Second * 40
+)
+
+const (
+	semaphoreName       = "expiry"
```

Comment:
Nit: This is too generic I feel, I would make it more specific to avoid potential conflicts.
```suggestion
	semaphoreName       = "auth.expiry"
```